src/compiler/scala/tools/nsc/classpath/AggregateClassPath.scala

```diff
@@ -72,7 +72,13 @@ case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
     getDistinctEntries(_.sources(inPackage))
 
   override private[nsc] def list(inPackage: String): ClassPathEntries = {
-    val (packages, classesAndSources) = aggregates.map(_.list(inPackage)).unzip
+    val (packages, classesAndSources) = aggregates.map { cp =>
+      try {
+        cp.list(inPackage)
+      } catch {
+        case _: java.io.IOException => ClassPathEntries(Seq(), Seq())
```

Comment:
My idea was to handle broken JARs the same way nonexistent JARs are handled. E.g. when there's a typo in the name of a JAR file the compiler silently ignores it -- I thought it made sense to do the same for broken JARs